libs/db.php

```diff
@@ -1,56 +1,83 @@
 <?php
-
 use MvLabs\Chocosite\Entity\Tavoletta;
-
+use MvLabs\Chocosite\Model\Giacenze;
+use MvLabs\Chocosite\Model\Segno;
 require __DIR__ . '/../vendor/autoload.php';
-
 function creaConnessionePDO() {
     // Nella realtà evitare di connettersi al db con l'utente "root".
     // E' preferibile creare un utente ad-hoc
     return new PDO('mysql:host=localhost;dbname=mvchocolates', 'root', 'mvlabs');
 }
-
 function inizializzaListaProdotti() {
     $db = creaConnessionePDO();
     $stmt =  $db->prepare('SELECT * FROM prodotti');
     $stmt->execute();
-
     return $stmt->fetchAll(PDO::FETCH_CLASS, Tavoletta::class);
 }
+function inizializzaGiacenze() {
+    $db = creaConnessionePDO();
+    $stmt = $db->prepare('SELECT prodotti.codice, prodotti.descrizione, giacenze.qta  FROM prodotti LEFT JOIN giacenze ON prodotti.codice=giacenze.codice;');
+    $stmt->execute();
+    return $stmt->fetchAll(PDO::FETCH_CLASS, Giacenze::class);
+}
+function setGiacenza(stdclass $movimenta,Segno $segno){
 
+    $db = creaConnessionePDO();
+    try {
+        $db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+        $db->beginTransaction();
+        // legge l'array e ricava le quantità da movimentare e i prodotti da aggiornare
+        //(ipotizzando che nella tabella giacenze ci siano tutti i prodotti (delegando tutto all'inserimento di nuovi prodotti))
+        foreach ($movimenta as $prodotto) {
```

Comment:
`$movimenta` è dichiarata come una classe, ma poi ci iteri... Gli oggetti in php sono effettivamente iterabili, ma non so se era quello che volevi effettivamente fare